pkg/server/commands/listusers/list_users_rpc.go

```diff
@@ -327,39 +321,71 @@ func (l *listUsersQuery) expandIntersection(
 	pool.WithCancelOnError()
 	pool.WithMaxGoroutines(int(l.resolveNodeBreadthLimit))
 
-	intersectionFoundUsersChan := make(chan *openfgav1.User, 1)
+	wildcardKey := fmt.Sprintf("%s:*", req.GetUserFilters()[0].GetType())
 
-	children := rewrite.Intersection.GetChild()
-	for _, childRewrite := range children {
-		copyChildRewrite := childRewrite
+	clauses := rewrite.Intersection.GetChild()
+	intersectionFoundUsersChans := make([]chan *openfgav1.User, len(clauses))
+	for i, rewrite := range clauses {
+		i := i
+		rewrite := rewrite
+		intersectionFoundUsersChans[i] = make(chan *openfgav1.User, 1)
 		pool.Go(func(ctx context.Context) error {
-			return l.expandRewrite(ctx, req, copyChildRewrite, intersectionFoundUsersChan)
+			return l.expandRewrite(ctx, req, rewrite, intersectionFoundUsersChans[i])
 		})
 	}
 
 	errChan := make(chan error, 1)
 
 	go func() {
 		err := pool.Wait()
-		close(intersectionFoundUsersChan)
+		for i := range intersectionFoundUsersChans {
+			close(intersectionFoundUsersChans[i])
+		}
 		errChan <- err
 		close(errChan)
 	}()
 
+	var mu sync.Mutex
+	var wg sync.WaitGroup
+	wg.Add(len(clauses))
+
+	wildcardCount := atomic.Uint32{}
 	foundUsersCountMap := make(map[string]uint32, 0)
```

Comment:
```suggestion
	foundUsersCountMap := make(map[string]uint32, 0) // GUARDED_BY(mu).
```

similar to https://github.com/openfga/openfga/blob/1fd71eb98377280f96708d1eaf23f1e66de13a96/pkg/storage/memory/memory.go#L110